pkg/monitors/vsphere/service/gateway.go

```diff
@@ -89,7 +81,14 @@ func (g *Gateway) queryPerfProviderSummary(ref types.ManagedObjectReference) (*t
 }
 
 func (g *Gateway) queryPerf(invObjs []*model.InventoryObject, maxSample int32) (*types.QueryPerfResponse, error) {
-	specs := make([]types.PerfQuerySpec, 0, len(invObjs))
+	numObjs := len(invObjs)
+	if numObjs == 0 {
+		// empty inventory, return empty response
+		// passing an empty spec to the api causes an error
+		return &types.QueryPerfResponse{}, nil
```

Comment:
Can you add a debug log statement informing the user that inventory is empty and we won't query vsphere ?

Reply:
Definitely.